test/smoketest_artman.py

```diff
@@ -122,14 +123,15 @@ def _parse(artman_yaml_path):
     return config_pb
 
 
-def _generate_artifact(artman_config, artifact_name, root_dir, log_file):
+def _generate_artifact(artman_config, artifact_name, root_dir, user_config, log_file):
```

Comment:
I don't want to be too picky about the test code but here the parameter order is weird, and 5 parameters is a lot :)
We can leave it as is for now (I'd move `user_config` closer to `artman_config` though), but if we ever get more parameters, we'll need to group them into objects. First link in Google: https://refactoring.guru/smells/long-parameter-list

Reply:
Ok, moved the `user_config` parameter closer to the `artman_config`.

Would passing the artman_config as a named arguments be better?

Reply:
I think names arguments would make code more readable. But again, since it's just a test code, we can leave it as is :)